source/gui/_addonStoreGui/controls/storeDialog.py

```diff
@@ -0,0 +1,337 @@
+		numInProgress = len(self._storeVM._downloader.progress)
+		if numInProgress:
+			res = gui.messageBox(
+				# Translators: Message shown prior to installing add-ons when closing the add-on store dialog
+				# The placeholder {} will be replaced with the number of add-ons to be installed
+				pgettext("addonStore", "Download of {} add-ons in progress, cancel downloading?").format(
+					numInProgress
+				),
+				self._installationPromptTitle,
+				style=wx.YES_NO
+			)
+			if res == wx.YES:
+				log.debug("Cancelling the download.")
+				self._storeVM.cancelDownloads()
+				# Continue to installation if any downloads completed
+			else:
+				# Let the user return to the add-on store and inspect add-ons being downloaded.
+				return
+
+		if self._storeVM._pendingInstalls:
+			installingDialog = gui.IndeterminateProgressDialog(
+				self,
+				self._installationPromptTitle,
+				# Translators: Message shown while installing add-ons after closing the add-on store dialog
+				# The placeholder {} will be replaced with the number of add-ons to be installed
+				pgettext("addonStore", "Installing {} add-ons, please wait.").format(len(self._storeVM._pendingInstalls))
+			)
+			self._storeVM.installPending()
+			wx.CallAfter(installingDialog.done)
+
+		# let the dialog exit.
+		super().onClose(evt)
+
+	@property
+	def _requiresRestart(self) -> bool:
+		if self._storeVM._pendingInstalls:
+			return True
+
+		for addonsForChannel in self._storeVM._installedAddons.values():
+			for addon in addonsForChannel.values():
+				if addon._addonHandlerModel.requiresRestart:
+					log.debug(f"Add-on {addon.name} modified, restart required")
+					return True
+
+		return False
+
+	@property
+	def _statusFilterKey(self) -> _StatusFilterKey:
+		index = self.addonListTabs.GetSelection()
+		return list(_statusFilters.keys())[index]
+
+	@property
+	def _channelFilterKey(self) -> Channel:
+		index = self.channelFilterCtrl.GetSelection()
+		return list(_channelFilters.keys())[index]
+
+	@property
+	def _titleText(self) -> str:
+		return f"{self.title} - {self._listLabelText}"
+
+	@property
+	def _listLabelText(self) -> str:
+		return f"{self._channelFilterKey.displayString} {self._statusFilterKey.displayString}"
+
+	def _setListLabels(self):
+		self.listLabel.SetLabelText(self._listLabelText)
+		self.SetTitle(self._titleText)
+
+	def _toggleFilterControls(self):
+		if self._storeVM._filteredStatusKey in {
+			_StatusFilterKey.AVAILABLE,
+			_StatusFilterKey.UPDATE,
+		}:
+			self._storeVM._filterChannelKey = Channel.STABLE
+			self.enabledFilterCtrl.Hide()
+			self.enabledFilterCtrl.Disable()
+			self.includeIncompatibleCtrl.Enable()
+			self.includeIncompatibleCtrl.Show()
+		else:
+			self._storeVM._filterChannelKey = Channel.ALL
+			self.enabledFilterCtrl.Show()
+			self.enabledFilterCtrl.Enable()
+			self.includeIncompatibleCtrl.Hide()
+			self.includeIncompatibleCtrl.Disable()
+
+	def onListTabPageChange(self, evt: wx.EVT_CHOICE):

```

Comment:
This method allows to update the same add-on multiple times, selecting the update action in the updatable add-ons page, moving to a different page and again there.

Reply:
As this is UX feedback, could you please write up a comment in https://github.com/nvaccess/nvda/discussions/14912.

Reply:
additionally - this code snippet is not the cause. We need to ensure the correct status is fetched when loading add-ons with an update, pending installation.